routers/api/v1/user/repo.go

```diff
@@ -44,7 +43,7 @@ func listUserRepos(ctx *context.APIContext, u *user_model.User, private bool) {
 			ctx.Error(http.StatusInternalServerError, "GetUserRepoPermission", err)
 			return
 		}
-		if ctx.IsSigned && ctx.Doer.IsAdmin || permission.UnitAccessMode(unit_model.TypeCode) >= perm.AccessModeRead {
+		if ctx.IsSigned && ctx.Doer.IsAdmin || permission.AccessMode >= perm.AccessModeRead {
```

Comment:
```suggestion
		if ctx.IsSigned && ctx.Doer.IsAdmin || permission.HasAnyUnitAccess() {
```

Reply:
Fixed (wasn't too familiar with how repo unit access works but it makes sense now after further reading)